prow/pod-utils/gcs/target.go

```diff
@@ -35,7 +35,7 @@ func PathForSpec(spec *downwardapi.JobSpec, pathSegment RepoPathBuilder) string
 	case kube.PeriodicJob, kube.PostsubmitJob:
 		return path.Join("logs", spec.Job, spec.BuildID)
 	case kube.PresubmitJob:
-		return path.Join("pr-logs", "pull", pathSegment(spec.Refs.Org, spec.Refs.Repo), strconv.Itoa(spec.Refs.Pulls[0].Number), spec.Job, spec.BuildID)
+		return path.Join("pr-logs", "pull", pathSegment(spec.Refs[0].Org, spec.Refs[0].Repo), strconv.Itoa(spec.Refs[0].Pulls[0].Number), spec.Job, spec.BuildID)
```

Comment:
Eeeeee. I am not a huge fan of these. I think we should just mirror the major ref and extra refs fields. Complex logic about job type --> how many possible entries are in the array seems error-prone and unchecked indexing is going to bite us at some point when some user finds some novel way to use this.

@BenTheElder @cjwagner PTAL